math/src/field/extensions/quadratic.rs

```diff
@@ -120,6 +121,11 @@ impl<B: StarkField> FieldElement for QuadExtensionA<B> {
         let result = B::prng_vector(seed, n * 2);
         Self::base_to_quad_vector(result)
     }
+
+    fn normalize(&mut self) {
```

Comment:
Why do we need mutability here and not deserialize/construct the elements directly in their canonical form? So no object can even be non-canonical in the first place.

Reply:
There are a couple of reasons:

First, when we use Montgomery reduction/multiplication, it is expensive to convert to and from Montgomery form all the time - so, we try to keep elements in Montgomery form as long as possible. This means that the internal representation of elements is in Montgomery form and we convert to/from Montgomery form only when an element is created or serialized.

The above doesn't require redundant representation of elements - every internal representation maps to a unique canonical representation. However, to improve efficiency we can encode elements in redundant form. That is, each canonical representation can have more than one internal representation (in the case of `f62` field, there are two internal representations per canonical representation since the range for internal elements is `[0, 2M)`). For example, this allows us to avoid the last conditional check in the multiplication procedure (so, multiplication becomes fully branchless).